### PR TITLE
Solution react_redux-list-of-posts

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**
[DEMO LINK](https://NazariiAlieksieiev.github.io/react_redux-list-of-posts/)

### CODE REVIEW

File: src/features/comments.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      state.loaded = false;
+    });
+
+    builder.addCase(init.rejected, (state) => {
+      state.loaded = false;
+      state.hasError = true;
+    });
+
+    builder.addCase(remove.fulfilled, (state, action) => {
+      const newCommets = state.comments
```
Comment (etojeDenys) [inline src/features/comments.ts:61]: There is a typo in the variable name 'newCommets'. Rename it to 'newComments'.

File: src/components/NewCommentForm.tsx [hunk 2]
```diff
@@ -59,14 +57,30 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
...
+
+      addNewComment();
+    }
 
-    // and the spinner will disappear immediately
-    setSubmitting(false);
     setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
-  };
+  }, [[name, email, body, selectedPost]]);
```
Comment (etojeDenys) [inline src/components/NewCommentForm.tsx:83]: Remove the double brackets in the dependency array. Change it to [name, email, body, selectedPost].

File: src/App.tsx [hunk 3]
```diff
@@ -76,18 +77,14 @@ export const App: React.FC = () => {
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length === 0 && (
+                {author && !loading && !error && posts.length === 0 && (
```
Comment (etojeDenys) [inline src/App.tsx:80]: it's convenient to name boolean variables as `isLoading`

File: src/features/comments.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  },
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(init.pending, (state) => {
+      state.loaded = true;
```
Comment (etojeDenys) [inline src/features/comments.ts:44]: better to rename it to `isLoading` instead of `loaded`

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by IvanFesenko — CHANGES_REQUESTED** [review]

File: src/components/PostsList.tsx [hunk 5]
```diff
@@ -1,56 +1,60 @@
...
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+  const buttonHandler = useCallback((post: Post) => {
+    return selectedPost?.id !== post.id
+      ? dispatch(setPost(post))
+      : dispatch(setPost(null));
+  }, [selectedPost]);
```
[the remark this answers is not in the document]
Comment (NazariiAlieksieiev) [reply src/components/PostsList.tsx:18]: @IvanFesenko eslint does not allow to do this

**Review by etojeDenys — APPROVED** [review]
lgtm

**Pull request closed without merging**